src/main/java/io/ballerina/plugins/idea/preloading/BallerinaLSPUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.plugins.idea.preloading;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import io.ballerina.plugins.idea.extensions.BallerinaLSPExtensionManager;
+import io.ballerina.plugins.idea.extensions.BallerinaLanguageServerDefinition;
+import io.ballerina.plugins.idea.sdk.BallerinaSdkService;
+import io.ballerina.plugins.idea.sdk.BallerinaSdkUtils;
+import org.jetbrains.annotations.NotNull;
+import org.wso2.lsp4intellij.IntellijLanguageClient;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_EXT_NAME;
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_LOG_PREFIX;
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_LS_CMD;
+
+/**
+ * Utility function for connecting with the Ballerina Language Server.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaLSPUtils {
+
+    private static final Set<Project> registeredProjects = new HashSet<>();
```

Comment:
As per the current implementation, it seems we are not updating this set when closing a project. Therefore LS will not be reconnected if we close and reopen a particular project. 

Also, do we need to keep the entire project objects in here? can't we keep only the project id to do the same thing?